src/test/java/com/synopsys/integration/jenkins/detect/substeps/DetectJarManagerTest.java

```diff
@@ -0,0 +1,107 @@
+package com.synopsys.integration.jenkins.detect.substeps;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mockito;
+
+import com.synopsys.integration.jenkins.extensions.JenkinsIntLogger;
+import com.synopsys.integration.log.LogLevel;
+
+import hudson.model.TaskListener;
+
+public class DetectJarManagerTest {
+
+    private final static String testDetectJarPath = "/test/detect/jar/path/detect.jar";
+    private final static String testJavaHome = "/test/java/home/path";
+    private final static String testPathEnv = "/test/path/env";
+
+    private final ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+    private final PrintStream originalOut = System.out;
+
+    @BeforeEach
+    public void setUp() {
+        System.setOut(new PrintStream(byteArrayOutputStream));
+    }
+
+    @AfterEach
+    public void tearDown() {
+        System.setOut(originalOut);
+    }
+
+    private static Stream<Arguments> setUpForExecutionInput() {
+        return Stream.of(
+            Arguments.of(null, true, null, testPathEnv, LogLevel.DEBUG),
+            Arguments.of(testJavaHome, false, testDetectJarPath, testPathEnv, LogLevel.WARN),
+            Arguments.of(testJavaHome, false, testDetectJarPath, testPathEnv, LogLevel.INFO),
+            Arguments.of(testJavaHome, false, testDetectJarPath, testPathEnv, LogLevel.DEBUG),
+            Arguments.of(testJavaHome, false, testDetectJarPath, testPathEnv, LogLevel.TRACE),
+            Arguments.of("", false, "", "", LogLevel.DEBUG),
+            Arguments.of(" ", false, " ", " ", LogLevel.DEBUG),
+            Arguments.of("", false, "", "", null)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource({ "setUpForExecutionInput" })
+    public void testSetUpForExecution(String javaHomeInput, Boolean isValidJavaHome, String detectJarPathInput, String pathEnvInput, LogLevel logLevel) {
```

Comment:
I'm not sure if this test should be a parameterized test. The test is largely lines 77 and 78, with a variable number of assertions on the lines following. Lines 62 through 75 all seem to be setup. Is this implementation preferable to a common method that performs the test and several different 6-7 line tests that are very clear in what they assert?